compiler/rustc_borrowck/src/diagnostics/mod.rs

```diff
@@ -1128,8 +1128,12 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
                                 "{place_name} {partially_str}moved due to this method call{loop_message}",
                             ),
                         );
+
                         let infcx = tcx.infer_ctxt().build();
```

Comment:
why do we even create a new infcx here instead of using `self.infcx`?

Reply:
Regions already solve so can_eq ices

Reply:
@compiler-errors suggested it in https://github.com/rust-lang/rust/pull/106095#discussion_r1056654341

Reply:
remember that someone told me the reason behind this before and I forgot :sweat_smile: 

might be nice to add a comment about this here, but apart from that this does feel like a pretty good solution :thinking: